### DISCUSSION TITLE
Config validator detects config.js file of an application that is not a runner

### DISCUSSION THREAD

**Discussion by vntw** [renovatebot/renovate#23095] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v35.78.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

Hey, I use the config validator for all repositories that use a renovate config via `npx --package renovate -c renovate-config-validator`. When adding renovate to an existing application/repository that already uses a `config.js`, then the validator reports errors because of an invalid config.

AFAIK the `config.js` is only used by the runner, so in my case it would be helpful if I could control whether the config.js should be validated or the exit code is >0. By default, I'd disable it since I only have one runner repository vs a lot of non-runners. The alternative is removing the config validator from that repository or changing the way the application works because of tooling conventions, both not ideal IMO :/

Maybe a new env param to control it or being able to set `RENOVATE_CONFIG_FILE='-'` or similar?

### What you tried so far.

I tried using the `RENOVATE_CONFIG_FILE` env param, but looking at https://github.com/renovatebot/renovate/blob/4dc6d691f60ae6c281f0d47e7930bfc938d484e7/lib/config-validator.ts#L117-L131, I can't get around the fact that if this param is set, a config must exist. So setting it to an invalid path or empty value only produces an exit code >0

Regarding the "It used to work …" - I'm not sure since I only now encountered the issue

### Relevant debug logs

<details><summary>Logs</summary>

```
 INFO: Validating config.js
ERROR: Found errors in configuration
       "file": "config.js",
       "errors": [
         {
           "topic": "Configuration Error",
           …
```

</details>

**Comment by rarkins** [comment, marked as answer]
It doesn't work to point it to renovate.json?

**Reply by vntw** [reply]
Oh good point 🤦  I, for some reason, thought the validation of the runner config would be different compared to the others but seems I was wrong. That way things get validated twice but no error anymore - good enough for me, thanks for the super quick hint 🙂
